### PR TITLE
Fix issue #292

### PR DESCRIPTION
**Pull request by ashwinrajesh8**
- Resolving Issue #292 
- Added support for need_extended_hours_data within the time interval shortcut functions.
- Default state of need_extended_hours_data set to True

Co-Authored-By: Ashwin Rajesh <ashwin.rajesh@gmail.com>

### CODE REVIEW

**Comment by alexgolec** [comment]
Please add tests

File: tests/client_test.py [hunk 1]
```diff
@@ -1293,6 +1373,46 @@ def test_get_price_history_every_ten_minutes_end_datetime(self):
...
+                'endDate': NOW_DATETIME_PLUS_SEVEN_DAYS_TIMESTAMP_MILLIS,
+                'needExtendedHoursData': True,
+        }
+        self.mock_session.get.assert_called_once_with(
+            self.make_url('/v1/marketdata/AAPL/pricehistory'),
+            params=params)
+
+
+    @patch('tda.client.base.datetime.datetime', mockdatetime)
+    def test_get_price_history_every_ten_minutes_vanilla_without_extendedhours(self):
```
Comment (alexgolec) [inline tests/client_test.py:1397]: `vanilla_without` is this intended?

File: tda/client/base.py [hunk 2]
```diff
@@ -819,7 +819,7 @@ def __normalize_start_and_end_datetimes(self, start_datetime, end_datetime):
 
 
     def get_price_history_every_minute(
-            self, symbol, *, start_datetime=None, end_datetime=None):
+            self, symbol, *, start_datetime=None, end_datetime=None, need_extended_hours_data=True):
```
Comment (alexgolec) [inline tda/client/base.py:822]: I should have been clear, let's leave the default value as `None` to be the same as the main function. Also please insert a line break to come in under 80 chars.

**Comment by alexgolec** [comment]
Note this change breaks tests.

**Comment by ashwinrajesh8** [comment]
Going through error logs indicated that most of the failed test cases were due to the absence of needExtendedHoursData in params, example below:
image
Adding this parameter to the tests should patch 42 of 56 failed cases, the remaining 14 have to do with the no_extendedhours tests.

**Comment by alexgolec** [comment]
Sounds good, please ping the review when your changes are fixed

File: tests/client_test.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import asyncio
 import datetime
 import os
+from tkinter import FALSE
```
Comment (alexgolec) [inline tests/client_test.py:4]: wat

File: tests/client_test.py [hunk 4]
```diff
@@ -1169,6 +1170,48 @@ def test_get_price_history_every_minute_end_datetime(self):
...
+                'endDate': NOW_DATETIME_PLUS_SEVEN_DAYS_TIMESTAMP_MILLIS,
+                'needExtendedHoursData': True,
+        }
+        self.mock_session.get.assert_called_once_with(
+            self.make_url('/v1/marketdata/AAPL/pricehistory'),
+            params=params)
+
+
+    @patch('tda.client.base.datetime.datetime', mockdatetime)
+    def test_get_price_history_every_minute_no_extendedhours(self):
```
Comment (alexgolec) [inline tests/client_test.py:1195]: I can't help but notice that this function is never actually called with the default parameters (i.e. `need_extended_hours_data=None`) This means that the most common path is actually never tested.

**Review by alexgolec — CHANGES_REQUESTED** [review]

File: tests/client_test.py (hunk 3 above)
Comment (ashwinrajesh8) [reply tests/client_test.py]: whoops lol

File: tests/client_test.py (hunk 4 above)
Comment (ashwinrajesh8) [reply tests/client_test.py:1214]: Yup, makes sense to add default value to tests. 

The `needExtendedHoursData` should no longer appear in expected response params when passing `None`, and td ameritrade api will set to true by default.

We could also just initialize `need_extended_hours_data=True` when we pass it into utility functions, however, the current implementation works fine as is.

Latest commit includes the tests for `need_extended_hours_data=None` scenarios.

File: tests/client_test.py [hunk 5]
```diff
@@ -1169,6 +1169,68 @@ def test_get_price_history_every_minute_end_datetime(self):
...
```
Comment (alexgolec) [inline tests/client_test.py:1214]: Optional: I feel like you can remove these `no_extendedhours` tests, given that they really only test that the appropriate value is passed in to the implementation method and not that any real logic is run. Feel free to leave them in, though. 

I'll merge the PR once you resolve this comment.

Comment (ashwinrajesh8) [reply tests/client_test.py]: Yea, guess you're right since `extendedhours` already demonstrates successful functionality of the new feature and `empty_extendedhours` covers the other edge case. `no_extendedhours` appears redundant.

**Comment by alexgolec** [comment]
LGTM, merging. Thanks for the PR!
